File: 251003/Habrus/src/main/java/by/bsuir/distcomp/dto/mapper/ArticleMapper.java
package by.bsuir.distcomp.dto.mapper;

import by.bsuir.distcomp.dto.request.ArticleRequestTo;
import by.bsuir.distcomp.dto.response.ArticleResponseTo;
import by.bsuir.distcomp.entity.Article;
import org.mapstruct.Mapper;

@Mapper(componentModel = "spring")
public interface ArticleMapper {
    ArticleResponseTo toDto(Article article);
    Article toEntity(ArticleRequestTo articleDTO);
}

File: 251002/Nikonorova/labs3-5/publisher/Mappers/ArticleMapper.cs
using publisher.DTO.Request;
using publisher.DTO.Response;
using Riok.Mapperly.Abstractions;
using publisher.Models;
using static System.Net.Mime.MediaTypeNames;

namespace publisher.Mappers
{
    [Mapper]
    public static partial class ArticleMapper
    {
        public static partial Article ToEntity(this CreateArticleRequestTo createArticleRequestTo);
        public static partial ArticleResponseTo ToResponse(this Article article);
        public static partial IEnumerable<ArticleResponseTo> ToResponse(this IEnumerable<Article> articles);
    }
}
